raster-foundry/raster-foundry
# Use metrics middleware for mosaic and analysis requests

## Overview

This PR uses the metric dao introduces in #4916 to count requests coming in for tiles from the project tile endpoints and the analysis endpoints.

### Checklist


### Notes

~[wip] until I make a real upsert out of the db interaction, I forgot those were a thing but without them the tile requests come in too simultaneously to end up with the right number of records~

## Testing Instructions

- reassemble backsplash
- run migrations
- fire up your server
- load some tiles
- select * from metrics
- there shouldn't be anything because the flag isn't on
- stop your server
- turn the flag to true in docker-compose (it's the last environment variable)
- load some tiles
- select * from metrics
- you should see something that makes sense -- a record with a value equal to your tile request count
- throw the z/x/y url into geojson.io
- you should get another record, because the referer is different 
- load some analysis tiles in the v2 ui and the lab
- you should get records for each of them, and the metadata in `metric_event` should make sense for them

Closes #4907
